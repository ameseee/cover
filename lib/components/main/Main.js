import React from 'react';

const Main = () => {

  return (
    <div>
      <nav>
        <h1>Cover</h1>
        <ul>
          <li>Manage Contacts</li>
          <li>Settings</li>
          <li>LogOut</li>
        </ul>
      </nav>
      <main>
        <section className="emergency-section">
          <h3>Emergency</h3>
          <button>TEXT 911</button>
        </section>

        <section className="contacts-section">
          {/* contact cards */}
        </section>
      </main>
    </div>
  );
};

export default Main;
