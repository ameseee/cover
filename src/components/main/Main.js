import React from 'react';
import { Link } from 'react-router-dom';
import { fetchScopedUsers } from './../../utils/fetchScopedUsers';
import PropTypes from 'prop-types';
import firebase from './../../firebase';
import ContactCards from '../../components/contactCards/ContactCards';
import { sendCustom } from '../../utils/postToServer';

class Main extends React.Component {
  constructor() {
    super();
    this.fetchScopedUsers = fetchScopedUsers.bind(this);
  }

  componentDidMount() {
    this.fetchScopedUsers(firebase);
  }

  handleSend = () => {
    sendCustom('911', '5756441355');
  }

  loadContacts = () => {
    const { loadedContacts } = this.props;

    return loadedContacts.map((contact, index) => {
      return (
        <ContactCards
          key={index}
          name={contact.contactName}
        />
      );
    });
  }

  noContacts() {
    return (
      <article className="no-contacts-card">
        <h4
          className="no-contacts-text">
          {"You don't have any contacts saved yet!"}
        </h4>
        <button className="add-contacts-btn">
          <Link
            className="add-contacts-link"
            to="/contacts"
          >
            Add Contacts
          </Link>
        </button>
      </article>
    );
  }

  render() {
    const { loadedContacts } = this.props;
    return (
      <div className="main">
        <main>
          <section className="emergency-section">
            <h3 className="emergency-title">Emergency</h3>
            <button
              className="emergency-btn"
              onClick={() => this.handleSend()}>TEXT 911</button>
          </section>
          <section className="hotline-section">
            <h3 className="hotline-title">NDV Hotline</h3>
            <button className="hotline-btn">Call Hotline</button>
          </section>
          <section className="contacts-section">
            <h3 className="section-title">Contacts</h3>
            { this.loadContacts() }
            { loadedContacts.length < 1
              ? this.noContacts()
              : null }
          </section>
        </main>
      </div>
    );
  }
}

Main.propTypes = {
  currentUser: PropTypes.string,
  loadContacts: PropTypes.func,
  loadedContacts: PropTypes.array,
};

export default Main;
