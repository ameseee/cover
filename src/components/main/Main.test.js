import React from 'react';
import Main from './Main';
import { shallow } from 'enzyme';

describe('<Main />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach( () => {
    const wrapper = shallow(<Main loadedContacts={[]}/>);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it.skip('should render a main section', () => {
    const main = wrapper.find('main');

    expect(main.length).toEqual(1);
  });

  it.skip('should render three section titles', () => {
    const sectionTitle = wrapper.find('h3');

    expect(sectionTitle.length).toEqual(3);
  });

  it.skip('should render an emergency section', () => {
    const emergency = wrapper.find('.emergency-section');

    expect(emergency.length).toEqual(1);
  });

  it.skip('should render an emergency button', () => {
    const emergencyBtn = wrapper.find('.emergency-btn');

    expect(emergencyBtn.length).toEqual(1);
  });

  it.skip('should render a hotline section', () => {
    const hotline = wrapper.find('.hotline-section');

    expect(hotline.length).toEqual(1);
  });

  it.skip('should render a hotline button', () => {
    const hotlineBtn = wrapper.find('.hotline-btn');

    expect(hotlineBtn.length).toEqual(1);
  });

  it.skip(
    'should render a "no contacts yet" card if the user has no contacts',
    () => {
      const noContacts = wrapper.find('.no-contacts-card');

      expect(noContacts.length).toEqual(1);
    });

  it.skip('should run a function if the user does have contacts', () => {
    const contactsWrapper =
    shallow(<Main loadedContacts={[{amy: '5756441355'}]}/>);

    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
