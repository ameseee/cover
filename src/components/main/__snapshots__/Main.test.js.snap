// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    loadedContacts={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <section
          className="emergency-section"
        >
          <h3
            className="emergency-title"
          >
            Emergency
          </h3>
          <button
            className="emergency-btn"
            onClick={[Function]}
          >
            TEXT 911
          </button>
        </section>
        <section
          className="hotline-section"
        >
          <h3
            className="hotline-title"
          >
            NDV Hotline
          </h3>
          <button
            className="hotline-btn"
          >
            Call Hotline
          </button>
        </section>
        <section
          className="contacts-section"
        >
          <h3
            className="section-title"
          >
            Contacts
          </h3>
          <article
            className="no-contacts-card"
          >
            <h4
              className="no-contacts-text"
            >
              You don't have any contacts saved yet!
            </h4>
            <button
              className="add-contacts-btn"
            >
              <Link
                className="add-contacts-link"
                replace={false}
                to="/contacts"
              >
                Add Contacts
              </Link>
            </button>
          </article>
        </section>
      </main>,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="emergency-section"
          >
            <h3
              className="emergency-title"
            >
              Emergency
            </h3>
            <button
              className="emergency-btn"
              onClick={[Function]}
            >
              TEXT 911
            </button>
          </section>,
          <section
            className="hotline-section"
          >
            <h3
              className="hotline-title"
            >
              NDV Hotline
            </h3>
            <button
              className="hotline-btn"
            >
              Call Hotline
            </button>
          </section>,
          <section
            className="contacts-section"
          >
            <h3
              className="section-title"
            >
              Contacts
            </h3>
            <article
              className="no-contacts-card"
            >
              <h4
                className="no-contacts-text"
              >
                You don't have any contacts saved yet!
              </h4>
              <button
                className="add-contacts-btn"
              >
                <Link
                  className="add-contacts-link"
                  replace={false}
                  to="/contacts"
                >
                  Add Contacts
                </Link>
              </button>
            </article>
          </section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="emergency-title"
              >
                Emergency
              </h3>,
              <button
                className="emergency-btn"
                onClick={[Function]}
              >
                TEXT 911
              </button>,
            ],
            "className": "emergency-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Emergency",
                "className": "emergency-title",
              },
              "ref": null,
              "rendered": "Emergency",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "TEXT 911",
                "className": "emergency-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "TEXT 911",
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="hotline-title"
              >
                NDV Hotline
              </h3>,
              <button
                className="hotline-btn"
              >
                Call Hotline
              </button>,
            ],
            "className": "hotline-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "NDV Hotline",
                "className": "hotline-title",
              },
              "ref": null,
              "rendered": "NDV Hotline",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Call Hotline",
                "className": "hotline-btn",
              },
              "ref": null,
              "rendered": "Call Hotline",
              "type": "button",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="section-title"
              >
                Contacts
              </h3>,
              Array [],
              <article
                className="no-contacts-card"
              >
                <h4
                  className="no-contacts-text"
                >
                  You don't have any contacts saved yet!
                </h4>
                <button
                  className="add-contacts-btn"
                >
                  <Link
                    className="add-contacts-link"
                    replace={false}
                    to="/contacts"
                  >
                    Add Contacts
                  </Link>
                </button>
              </article>,
            ],
            "className": "contacts-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Contacts",
                "className": "section-title",
              },
              "ref": null,
              "rendered": "Contacts",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="no-contacts-text"
                  >
                    You don't have any contacts saved yet!
                  </h4>,
                  <button
                    className="add-contacts-btn"
                  >
                    <Link
                      className="add-contacts-link"
                      replace={false}
                      to="/contacts"
                    >
                      Add Contacts
                    </Link>
                  </button>,
                ],
                "className": "no-contacts-card",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "You don't have any contacts saved yet!",
                    "className": "no-contacts-text",
                  },
                  "ref": null,
                  "rendered": "You don't have any contacts saved yet!",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="add-contacts-link"
                      replace={false}
                      to="/contacts"
                    >
                      Add Contacts
                    </Link>,
                    "className": "add-contacts-btn",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Add Contacts",
                      "className": "add-contacts-link",
                      "replace": false,
                      "to": "/contacts",
                    },
                    "ref": null,
                    "rendered": "Add Contacts",
                    "type": [Function],
                  },
                  "type": "button",
                },
              ],
              "type": "article",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <section
            className="emergency-section"
          >
            <h3
              className="emergency-title"
            >
              Emergency
            </h3>
            <button
              className="emergency-btn"
              onClick={[Function]}
            >
              TEXT 911
            </button>
          </section>
          <section
            className="hotline-section"
          >
            <h3
              className="hotline-title"
            >
              NDV Hotline
            </h3>
            <button
              className="hotline-btn"
            >
              Call Hotline
            </button>
          </section>
          <section
            className="contacts-section"
          >
            <h3
              className="section-title"
            >
              Contacts
            </h3>
            <article
              className="no-contacts-card"
            >
              <h4
                className="no-contacts-text"
              >
                You don't have any contacts saved yet!
              </h4>
              <button
                className="add-contacts-btn"
              >
                <Link
                  className="add-contacts-link"
                  replace={false}
                  to="/contacts"
                >
                  Add Contacts
                </Link>
              </button>
            </article>
          </section>
        </main>,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="emergency-section"
            >
              <h3
                className="emergency-title"
              >
                Emergency
              </h3>
              <button
                className="emergency-btn"
                onClick={[Function]}
              >
                TEXT 911
              </button>
            </section>,
            <section
              className="hotline-section"
            >
              <h3
                className="hotline-title"
              >
                NDV Hotline
              </h3>
              <button
                className="hotline-btn"
              >
                Call Hotline
              </button>
            </section>,
            <section
              className="contacts-section"
            >
              <h3
                className="section-title"
              >
                Contacts
              </h3>
              <article
                className="no-contacts-card"
              >
                <h4
                  className="no-contacts-text"
                >
                  You don't have any contacts saved yet!
                </h4>
                <button
                  className="add-contacts-btn"
                >
                  <Link
                    className="add-contacts-link"
                    replace={false}
                    to="/contacts"
                  >
                    Add Contacts
                  </Link>
                </button>
              </article>
            </section>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="emergency-title"
                >
                  Emergency
                </h3>,
                <button
                  className="emergency-btn"
                  onClick={[Function]}
                >
                  TEXT 911
                </button>,
              ],
              "className": "emergency-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Emergency",
                  "className": "emergency-title",
                },
                "ref": null,
                "rendered": "Emergency",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "TEXT 911",
                  "className": "emergency-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "TEXT 911",
                "type": "button",
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="hotline-title"
                >
                  NDV Hotline
                </h3>,
                <button
                  className="hotline-btn"
                >
                  Call Hotline
                </button>,
              ],
              "className": "hotline-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "NDV Hotline",
                  "className": "hotline-title",
                },
                "ref": null,
                "rendered": "NDV Hotline",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Call Hotline",
                  "className": "hotline-btn",
                },
                "ref": null,
                "rendered": "Call Hotline",
                "type": "button",
              },
            ],
            "type": "section",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  className="section-title"
                >
                  Contacts
                </h3>,
                Array [],
                <article
                  className="no-contacts-card"
                >
                  <h4
                    className="no-contacts-text"
                  >
                    You don't have any contacts saved yet!
                  </h4>
                  <button
                    className="add-contacts-btn"
                  >
                    <Link
                      className="add-contacts-link"
                      replace={false}
                      to="/contacts"
                    >
                      Add Contacts
                    </Link>
                  </button>
                </article>,
              ],
              "className": "contacts-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Contacts",
                  "className": "section-title",
                },
                "ref": null,
                "rendered": "Contacts",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="no-contacts-text"
                    >
                      You don't have any contacts saved yet!
                    </h4>,
                    <button
                      className="add-contacts-btn"
                    >
                      <Link
                        className="add-contacts-link"
                        replace={false}
                        to="/contacts"
                      >
                        Add Contacts
                      </Link>
                    </button>,
                  ],
                  "className": "no-contacts-card",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "You don't have any contacts saved yet!",
                      "className": "no-contacts-text",
                    },
                    "ref": null,
                    "rendered": "You don't have any contacts saved yet!",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="add-contacts-link"
                        replace={false}
                        to="/contacts"
                      >
                        Add Contacts
                      </Link>,
                      "className": "add-contacts-btn",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Add Contacts",
                        "className": "add-contacts-link",
                        "replace": false,
                        "to": "/contacts",
                      },
                      "ref": null,
                      "rendered": "Add Contacts",
                      "type": [Function],
                    },
                    "type": "button",
                  },
                ],
                "type": "article",
              },
            ],
            "type": "section",
          },
        ],
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
